sdk/python/v1beta1/kubeflow/katib/utils/utils.py

```diff
@@ -129,3 +132,132 @@ class FakeResponse:
 
     def __init__(self, obj):
         self.data = json.dumps(obj)
+
+
+class SetEncoder(json.JSONEncoder):
+    def default(self, obj):
+        if isinstance(obj, set):
+            return list(obj)
+        if isinstance(obj, type):
+            return obj.__name__
+        return json.JSONEncoder.default(self, obj)
+
+
+def parameter_substitution(
+    parameters: Union[Dict[str, Any], "TrainingArguments", "LoraConfig"],  # noqa: F821
+    experiment_params: List[models.V1beta1ParameterSpec],
+    trial_params: List[models.V1beta1TrialParameterSpec],
+):
+    if isinstance(parameters, dict):
```

Comment:
I think, if this function doesn't share any common code, it makes sense to separate it between 2 helper function.
In that case, you don't need to introduce if/else statement here.
For example, you can create 2 functions:
```
get_trial_substitutions_from_dict()
get_trial_substitutions_from_trainer()
```
WDYT @helenxie-bit ?

Reply:
Yeah, it makes sense. I will adjust accordingly.